hackathon_site/hardware/views.py

```diff
@@ -118,3 +120,31 @@ def post(self, request, *args, **kwargs):
             return HttpResponseServerError()
         response_data = response_serializer.data
         return Response(response_data, status=status.HTTP_201_CREATED)
+
+
+class OrderDetailView(generics.GenericAPIView, mixins.UpdateModelMixin):
+    queryset = Order.objects.all()
+    serializer_class = OrderChangeSerializer
+    permission_classes = [FullDjangoModelPermissions]
+
```

Comment:
See above, by moving all this logic into the serializer you can reuse it and have a much simpler view! No need for the `check_object_permissions` method on this view, since there aren't any object permissions in question.

Reply:
for sure, def cleaner